packages/rpc/src/index.ts

```diff
@@ -62,89 +62,83 @@ export const Code = component('rpc' as Component, {
   Unauthorized: '' as StatusCode,
   Forbidden: '' as StatusCode,
   BadRequest: '' as StatusCode,
-  UnknownMethod: '' as StatusCode<{method: string}>
+  UnknownMethod: '' as StatusCode<{ method: string }>
 })
 
 export interface RequestStream {
-  send: <T extends any[]>(req: Request<T>) => void
+  send: (request: Request<any>) => void
 }
 
 interface Operation {
-  id: number
-  request: () => void // Perform request again
+  id: ReqId
   promise: Promise<any>
-  resolve: <T> (value: T) => void
+  resolve: <T>(value: T) => void
+  reject: (reason?: any) => void
 }
 
 /**
- * Process RPC request and handle responses
+ * Process requests and handle responses.
+ * Also allow to handle non identified results passed from other side.
+ *
+ * Hold operations in progress and allow to retry them if required.
  */
-export class RequestProcessor implements Storage {
-  reqId: number = 0
+export class RequestProcessor {
+  reqIndex: number = 0
   requests = new Map<ReqId, Operation>()
   stream: () => Promise<RequestStream>
-  handler: (tx: Tx) => void
 
-  constructor (stream: () => Promise<RequestStream>, handler: (tx: Tx) => void) {
+  constructor (stream: () => Promise<RequestStream>) {
```

Comment:
Can we have `stream` to be `RequestStream`? And if we can, I believe using `(request: Request<any>) => void` is better than RequestStream -- no need to look what `RequestStream` means. Also, why not `(request: Request<any>) => Promise<void>`?